my_enumerable.rb

```diff
@@ -0,0 +1,272 @@
+# frozen_string_literal: true
+
+# Arraycompleted: all? any? include? none?, map, size, count, length, find_index
+# select, find, find_all, reject, max, min
+#
+# Hash:
+# not completed: all? any? include? none? each map! map
+# find_all find_index reject max min size count length select find
+
+module MyEnumerable
+  def my_all?(value = nil)
+    case
+    when value.nil? && !block_given?
+      each do |item|
+        return false unless item
+      end
+      true
+    when !value.nil?
+      each do |item|
+        return false unless value === item
+      end
+      true
+    when block_given?
+      each do |item|
+        return false unless yield(item)
+      end
+      true
+    else
+      "Error: value has an invalid value (#{value})"
+    end
```

Comment:
need refactoring. Too complicate....

Reply:
make sense to move logic to few methods and avoid use redundant switch case